### ISSUE TITLE
Tracking: Upgrade shared ECC dependencies and zcash_script for zcashd 5.6.0

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#6859]
## Motivation

We need to do a routine update to our shared ECC dependencies in Zebra:
- [x] #6836 
- [x] https://github.com/ZcashFoundation/zebra/pull/6855
- [x] #7107 
- [x] #6857
- [x] https://github.com/ZcashFoundation/zebra/pull/6854
- [x] https://github.com/ZcashFoundation/zebra/pull/6851
- [x] #7151
- [x] #6638

These should be done together to avoid duplicate dependencies.
We might want to do incrementalmerkletree in a separate PR because it has breaking API changes.

### Testing

Our existing tests cover these crates, except for the note commitment tree serialisation format in RPCs. That will need a test, ideally a snapshot test that is written based on the format before the upgrade.

## Related Work

This probably needs to happen before the RPC changes, but after the first stable release.

**Comment by mpguerra** [comment]
~~Part of this ticket is blocked by the zcashd 5.6.0 release~~

I see it's already been released 😅

**Comment by mpguerra** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/6859?planning-poker) with Zenhub @arya2 @dconnolly @oxarbitrage @teor2345 @upbqdn

**Comment by teor2345** [comment]
Estimation note: there are some API changes in `incrementalmerkletree`, see PR #6836 and maybe ticket #6642 for details.

**Comment by dconnolly** [comment]
@oxarbitrage for upgrading `incrementalmerkletree`, here's a summary of the changes I spotted as we upgrade to 0.4.0

https://github.com/ZcashFoundation/zebra/pull/6836#issuecomment-1579444904

**Comment by teor2345** [comment]
`zcash_proofs` updated to 0.12.1 (#7107)

**Comment by mpguerra** [comment]
@oxarbitrage do you want to update this estimate? :D

**Comment by oxarbitrage** [comment]
> @oxarbitrage do you want to update this estimate? :D

Changed to 8.
